tests/test_download.py

```diff
@@ -195,78 +224,268 @@ def test_download_masked_json_reproject(self, tmp_path):
+    @pytest.mark.parametrize("space_type", ["spacetime", "spatial"])
+    @pytest.mark.parametrize("format_arg", ["NETCDF", "GTIFF", "PNG"])
+    def test_write_assets_parameterize(self, tmp_path,
+                                       tiled,
+                                       stitch,
+                                       catalog,
+                                       tile_grid,
+                                       sample_by_feature,
+                                       batch_mode,
+                                       filename_prefix,
+                                       space_type,
+                                       format_arg,
+                                       ):
+        d = locals()
+        d = {i: d[i] for i in d if i != 'self' and i != "tmp_path" and i != "d"}
+        test_name = "-".join(map(str, list(d.values())))  # a bit like how pytest names it
+        if batch_mode and sample_by_feature:
+            # 'sample_by_feature' is only relevant in 'batch_mode'
+            return
+
+        if space_type == "spacetime":
+            input_layer = self.create_spacetime_layer()
+        else:
+            input_layer = self.create_spatial_layer()
+        imagecollection = GeopysparkDataCube(pyramid=gps.Pyramid({0: input_layer}))
+        imagecollection.metadata = imagecollection.metadata.add_dimension('band_one', 'band_one', 'bands')
+        imagecollection.metadata = imagecollection.metadata.append_band(Band('band_two', '', ''))
 
+        if format_arg == "NETCDF":
+            extension = ".nc"
+        elif format_arg == "PNG":
+            extension = ".png"
+        elif format_arg == "GTIFF":
+            extension = ".tif"
+        else:
+            assert False
+        filename = "test_download_result" + extension
+        assets = imagecollection.write_assets(
+            str(tmp_path / filename),
+            format=format_arg,
+            format_options={
+                "tiled": tiled,
+                "stitch": stitch,
+                "parameters": {
+                    "catalog": catalog,
+                },
+                "tile_grid": tile_grid,
+                "sample_by_feature": sample_by_feature,
+                "batch_mode": batch_mode,
+                "filename_prefix": filename_prefix,  # no effect when outputting single file
+
+                # non parametrized:
+                "geometries": geojson_to_geometry(self.features),
+                # "feature_id_property": 'id',  # not used
+                # "multidate": True,  # not used
+            }
+        )
+        # with open(self.test_write_assets_parameterize_path + test_name + ".json", 'w') as fp:
+        #     json.dump(assets, fp, indent=2)
+
+        name, asset = next(iter(assets.items()))
+        print("href of first asset: " + asset['href'])
+
+        # Special case for saveRDDTemporal as it only returns one case, but is designed to return multiple cases
```

Comment:
@jdries Here I was not sure why `saveRDDTemporal` only returns one file, while other code paths returned multiple files.
This is not something I changed, but only added coverage for. So, ok to just document it like this?

Reply:
from what I can tell on sight, create_spacetime_layer creates a layer with only one date (self.now). Hence I would indeed inspect only one asset.
Test coverage would indeed be better if we have more than one date, can we change that method?

Reply:
That was indeed it. The hacky if-statement is removed now.